kiota_http/middleware/url_replace_handler.py

```diff
@@ -30,12 +31,17 @@ async def send(
         Returns:
             Response: The response object.
         """
+        response = None
+        _enable_span = self._create_observability_span(request, "UrlReplaceHandler_send")
+        _enable_span.set_attribute("com.microsoft.kiota.handler.url_replacer.enable", True)
         current_options = self._get_current_options(request)
 
         url_string: str = str(request.url)  #type: ignore
         url_string = self.replace_url_segment(url_string, current_options)
```

Comment:
the replacement and the span attribute being set to true should only happen if the option is enabled.